src/logcollector/config.c

```diff
@@ -195,6 +198,19 @@ void _getLocalfilesListJSON(logreader *list, cJSON *array, int gl) {
             cJSON_AddNumberToObject(multiline, "timeout", list[i].multiline->timeout);
             cJSON_AddItemToObject(file, "multiline_regex", multiline);
         }
+        if (list[i].regex_ignore) {
+            OSList_foreach(node_it, list[i].regex_ignore) {
+                dir_it = node_it->data;
+                cJSON_AddStringToObject(file, "ignore", w_expression_get_regex_pattern(dir_it));
```

Comment:
We are just including the regex value but we are missing the type. For every ignore and restrict we have to add a new object/item with the following:
```
{"ignore": [
  {"value": "regex1",
    "type": "PCRE2"
  },
  {"value": "regex2",
    "type": "OSRegex"
  },
  ...
  ]
}
```